### PR TITLE
Adds onRowDidClose event

### LINKED ISSUE #66: Add a lifecycle hook for onRowDidClose, etc...
This would be useful for applying computationally heavy stuff _after_ the row close is done animating.

### PR DESCRIPTION
**Pull request by levitanong**
Fixes #66

### CODE REVIEW

File: components/SwipeRow.js [hunk 1]
```diff
@@ -164,13 +164,16 @@ class SwipeRow extends Component {
...
+		Animated.spring(
+      this.state.translateX,
 			{
 				toValue,
 				friction: this.props.friction,
 				tension: this.props.tension
 			}
-		).start();
+		).start(function(){
+      this.props.onRowDidClose && this.props.onRowDidClose();
```
Comment (jemise111) [inline components/SwipeRow.js]: It looks like this is being called whenever a row is manuallySwiped. So it's being triggered whether a row is openedOrClosed. I'd suggest checking the `toValue` and seeing if it's `0` before calling `onRowDidClose()`

**Comment by levitanong** [comment]
@jemise111 no prob! Left you a comment on your review comment, but given that I've already made changes, it might get buried. Will repost here to make sure you see it, and for easy access.

> Oops, working on that. In testing my work though, I realized that onRowClose and onRowOpen do not work on SwipeRow (i.e. when renderHiddenRow is undefined,and renderRow is supplied with SwipeRow)
> 
> I'm guessing your intent is to make the onRowClose and onRowOpen defined on the SwipeListView pass those functions on to the SwipeRow, but perhaps if they are defined explicitly in the SwipeRow, they should be take precedence?

**Comment by jemise111** [comment]
@levitanong Thanks for making the changes! The formatting got a little messed up but I'm going to merge this and fix it so I can put this in the next release.

As for your other comment, yeah there has been a little confusion about how to handle onRow callbacks when you use a `SwipeListView` AND a `SwipeRow`. It would be possible to just trigger both callbacks but I haven't had time to implement yet but if you wanted to give it a shot that would be awesome.. :)

**Comment by levitanong** [comment]
@jemise111 Sorry about the formatting thing! I've been using spacemacs, and I have no idea how to deal with the formatting issues. For the record, how many tabs/spaces do you allot per indentation?

I can give triggering both callbacks a shot :)

**Comment by jemise111** [comment]
No worries on the formatting. I think it's just normal tabs. Just check the diff and make sure it all looks aligned. The PR looked like this:

image

so something got a bit messed up.

Good luck with the other PR, looking forward to it! :)

**Pull request merged**